src/socket.cpp

```diff
@@ -241,57 +266,73 @@ void CSocket::SendPacket ( const CVector<uint8_t>& vecbySendBuf, const CHostAddr
         // char vector in "const char*", for this we first convert the const
         // uint8_t vector in a read/write uint8_t vector and then do the cast to
         // const char *)
-        if ( HostAddr.InetAddr.protocol() == QAbstractSocket::IPv4Protocol )
+
+        for ( int tries = 0; tries < 2; tries++ ) // retry loop in case send fails on iOS
```

Comment:
Wouldn't an ifdef for iOS be better here (if it's only needed on iOS)?

Reply:
It's @softins code. I don't think iOS needs it though, but it might be a good idea to leave the loop there for all OSes, just as a safety mesure.

Reply:
I thought the whole point of using UDP was that it was lossy. How did a retry mechanism end up being needed here?

Reply:
@emlynmac in iOS when you lock the device, all inactive sockets are closed. When device is unlocked, and user tries to connect to a server for example, the app crashes because of the closed socket. 
As far as I understand, the status check in the loop only concerns the socket (because it can not know if packet reached its destination), if socket is closed, then status < 0, that's when we need to reinit the socket and send the packet again.

Reply:
@emlynmac  Application level reliability is common for ensuring a message gets through. However, for audio/video stream, retry/retransmission is not used.

Reply:
> @emlynmac Application level reliability is common for ensuring a message gets through. However, for audio/video stream, retry/retransmission is not used.

This is handled elsewhere for messages that need it.

@ngocdh Thanks - I see.